examples/sphere/shallow_water.jl

```diff
@@ -171,6 +184,78 @@ if test_name == rossby_haurwitz_test_name
             Geometry.UVVector(uλ, uϕ),
             local_geometry,
         )
+        return (h = h, u = u)
+    end
+elseif test_name == steady_state_compact_test_name
+    y0 = map(Fields.local_geometry_field(space)) do local_geometry
+        coord = local_geometry.coordinates
+
+        ϕ = coord.lat
+        λ = coord.long
+
+        if α == 0.0
+            ϕprime = ϕ
+            λprime = λ
+        else
+            ϕprime = asind(sind(ϕ) * cosd(α) - cosd(ϕ) * cosd(λ) * sind(α))
+            λprime = asind(sind(λ) * cosd(ϕ) / cosd(ϕprime))
+
+            # Temporary angle to ensure λprime is in the right quadrant
+            λcond = cosd(α) * cosd(λ) * cosd(ϕ) + sind(α) * sind(ϕ)
+
+            # If λprime is not in the right quadrant, adjust
+            if λcond < 0.0
+                λprime = -λprime - 180.0
+            end
+            if λprime < -180.0
+                λprime += 360.0
+            end
+        end
+
+        # Set auxiliary function needed for initial state of velocity field
+        b(x) = x ≤ 0.0 ? 0.0 : exp(-x^(-1))
+
+        x(ϕprime) = xₑ * (ϕprime - ϕᵦ) / (ϕₑ - ϕᵦ)
+        uλprime(ϕprime) =
+            u0 * b(x(ϕprime)) * b(xₑ - x(ϕprime)) * exp(4.0 / xₑ)
+        uϕprime = 0.0
+
+        # Set integral needed for height initial state
+        hᵢₙₜ(γ) =
+            abs(γ) < 90.0 ?
+            (2 * Ω * sind(γ) + uλprime(γ) * tand(γ) / R) * uλprime(γ) : 0.0
+
+        # Set initial state for height field
+        h = h0 - (R / g) * quadgk(γ -> hᵢₙₜ(γ), -90.0, ϕprime)[1]
```

Comment:
the anonymous function here is redundant:
```suggestion
        h = h0 - (R / g) * quadgk(hᵢₙₜ, -90.0, ϕprime)[1]
```